assets/app/scripts/controllers/newfromtemplate.js

```diff
@@ -0,0 +1,161 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:NewFromTemplateController
+ * @description
+ * # NewFromTemplateController
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('NewFromTemplateController', function ($scope, $http, $routeParams, DataService, $q, $location, TaskList, $parse) {
+
+    function errorPage(message) {
+      var redirect = URI('/error').query({
+      	error_description: message,
+      }).toString();
+      $location.url(redirect);
+    }
+    
+    var dcContainers = $parse('template.controllerTemplate.podTemplate.desiredState.manifest.containers');
+    var stiBuilderImage = $parse('parameters.strategy.stiStrategy.image');
+    var outputImage = $parse('parameters.output.to.name || parameters.output.DockerImageReference');
+    
+    function deploymentConfigImages(dc) {
+      var images = [];
+      var containers = dcContainers(dc);
+      if (containers) {
+        containers.forEach(function(container) {
+          images.push(container.image);
+        });
+      }
+      return images;
+    }
+
+    function imageItems(data) {
+      var images = [];
+      var dcImages = [];
+      var outputImages = {};
+      data.items.forEach(function(item) {
+        if (item.kind == "BuildConfig") {
+          var builder = stiBuilderImage(item)
+          if(builder) {
+            images.push({ name: builder });
+          }
+          var output = outputImage(item);
+          if (output) {
+            outputImages[output] = true;
+          }
+        }
+        if (item.kind == "DeploymentConfig") {
+          dcImages = dcImages.concat(deploymentConfigImages(item));
+        }
+      });
+      dcImages.forEach(function(image) {
+        if (!outputImages[image]) {
+          images.push({ name: image });
+        }
+      });
+      return images;
+    }
+    
+    $scope.createFromTemplate = function() { 
+      DataService.create("templateConfigs", $scope.template, $scope).then(
+        function(config) { // success
+          var titles = { 
+            started: "Creating " + $scope.template.metadata.name + " in project " + $scope.projectName,
```

Comment:
projectName is not the pretty name for the project, you should use `$scope.project.displayName || $scope.projectName`  (since displayName is optional)